Support playground during CHECKOUT_ROOT detection

dlam opened the pull request:
Currently, we hard-code "../.." to get to the root of git checkout for
declaring a location for build outputs, however this causes issues in
Github checkouts which live directly in ./androidx as opposed
to ./frameworks/support.

This changes the hard-coded "../.." to instead use git to parse the root
checkout directory.

Test: Manually - Gradle sync within Studio from github checkout

dlam committed:
Support playground during CHECKOUT_ROOT detection

Currently, we hard-code "../.." to get to the root of git checkout for
declaring a location for build outputs, however this causes issues in
Github checkouts which live directly in ./androidx as opposed
to ./frameworks/support.

This changes the hard-coded "../.." to instead use git to parse the root
checkout directory.

Test: Manually - Gradle sync within Studio from github checkout

dlam commented:
This fixes gradle sync when opening project from playground Studio from github checkout

dlam commented:
maybe we could ask jeff to review after we mirror into aosp?

yigit commented on buildSrc/out.gradle:
```diff
@@ -25,7 +25,8 @@ def chooseOutDir(subdir = "") {
         def checkoutRoot = System.getProperty("CHECKOUT_ROOT")
         if (checkoutRoot == null) {
-            checkoutRoot = new File("${buildscript.sourceFile.parent}/../../..")
+            def workingDir = new File(buildscript.sourceFile.parent)
```
not sure how this works. git repo is the same in the repo tool checkout as well.
we might consider hardcoding parent folders and cheeck for frameworks/support

dlam replied:
Ah yeah I may have mispoke here, the git repo root is the same, but this returns the repo checkout root as its relative of buildSrc/ but this leads outside the checkout in github

dlam replied:
Done

dlam committed:
Preserve existing behavior in out.gradle for non-github checkouts

This preserves using the repo checkout root if repo exists, otherwise
uses the git checkout root for github checkouts.

Test: cd paging && ./gradlew createArchive

dlam committed:
Catch IOException when repo is not installed
